plugins/CoreAdminHome/tests/Integration/ConfigDeleteTest.php

```diff
@@ -0,0 +1,425 @@
+
+        // Add a second section so we are testing that we do not accidentally return it.
+        $sectionName = self::TEST_SECTION_1_NAME . '_second_section';
+        // Add a setting with the same name as in section #1 but with a random int value.
+        $config->$sectionName[self::TEST_SETTING_1_1_NAME] = random_int(PHP_INT_MIN, PHP_INT_MAX);
+        // Add another setting to the same section with some bogus content.
+        $config->$sectionName[self::TEST_SETTING_1_2_NAME . '_another'] = '127.0.0.1';
+
+        // Add an array value like section=PluginsInstalled; setting=PluginsInstalled[].
+        $sectionName = self::TEST_SECTION_2_NAME;
+        // Add some values to the setting array.
+        $config->$sectionName[self::TEST_SETTING_2_1_NAME] = self::TEST_SETTING_2_1_VALUES;
+
+        $config->forceSave();
+        return $config;
+    }
+
+    private static function removeTestConfigFile()
+    {
+        $configPath = self::getTestConfigFilePath();
+        if (file_exists($configPath)) {
+            unlink($configPath);
+        }
+    }
+
+    private static function makeNewConfig()
+    {
+        $settings = new GlobalSettingsProvider(null, self::getTestConfigFilePath());
+        return new Config($settings);
+    }
+
+    private function runCommandWithOptions(string $sectionName, string $settingName, string $value = ''): object
+    {
+        $debug = false;
+        $debug && fwrite(STDERR, PHP_EOL . str_repeat('-', 80) . PHP_EOL . __FUNCTION__ . "::Started with \$sectionName=$sectionName; \$settingName=$settingName; \$value=$value") . PHP_EOL;
+
+        $inputArr = [
+            'command' => self::COMMAND,
+            '--section' => $sectionName,
+            '--key' => $settingName,
+            '-vvv' => false,
+        ];
+        if ( ! empty($value)) {
+            $inputArr['--value'] = $value;
+        }
+        $exitCode = $this->applicationTester->run($inputArr);
+
+        // Pass true to getDisplay(true) to normalize line endings, then trim() bc CLI adds an \ automatically.
+        $output = trim($this->applicationTester->getDisplay(true));
+        $debug && fwrite(STDERR, PHP_EOL . __FUNCTION__ . '::Got command output=' . $output) . PHP_EOL;
+
+        // Put the results in an easy-to-handle object format.
+        return (object) ['exitCode' => $exitCode, 'output' => $output];
+    }
+
+    private function runCommandWithArguments(string $sectionName, string $settingName = '', string $value = ''): object
+    {
+        $debug = false;
+        $debug && fwrite(STDERR, PHP_EOL . str_repeat('-', 80) . PHP_EOL . __FUNCTION__ . "::Started with \$sectionName=$sectionName; \$settingName=$settingName; \$value=$value") . PHP_EOL;
+
+        $inputArr = [
+            'command' => self::COMMAND,
+            '-vvv' => false,
+            'argument' => $sectionName . $settingName . (empty($value) ? '' : ".$value"),
```

Comment:
This would actually need to be this in order to work:
```suggestion
            'argument' => $sectionName . '.' . $settingName . (empty($value) ? '' : ".$value"),
```

Reply:
Good catch.  Fixed.